presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/AbstractSqlInvokedFunctionNamespaceManager.java

```diff
@@ -240,8 +258,17 @@ protected FunctionImplementationType getFunctionImplementationType(SqlInvokedFun
     protected ScalarFunctionImplementation sqlInvokedFunctionToImplementation(SqlInvokedFunction function)
     {
         FunctionImplementationType implementationType = getFunctionImplementationType(function);
-        checkArgument(implementationType.equals(SQL));
-        return new SqlInvokedScalarFunctionImplementation(function.getBody());
+        switch (implementationType) {
+            case SQL:
+                return new SqlInvokedScalarFunctionImplementation(function.getBody());
+            case THRIFT:
+                checkArgument(function.getFunctionHandle().isPresent(), "Need functionHandle to get function implementation");
+                return new ThriftScalarFunctionImplementation(function.getFunctionHandle().get(), function.getRoutineCharacteristics().getLanguage());
+            case BUILTIN:
+                throw new IllegalStateException(format("%s cannot manage BUILTIN functions", this.getClass()));
```

Comment:
Since the method name is called `sqlInvokedFunctionToImplementation`, a better error message could be something like
```
SqlInvokedfunction xx cannot be BUILTIN
SqlInvokedfunction xx cannot have BUILTIN implementation type
```

Reply:
But the reason it cannot be BUILTIN is because this function namespace manager cannot manage builtin function right?

Reply:
Added the function name as well.